src/API/Google/Query/MerchantPriceSuggestionsQuery.php

```diff
@@ -0,0 +1,62 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Query;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Query\MerchantQuery;
+
+/**
+ * Class MerchantPriceSuggestionsQuery
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Query
+ */
+class MerchantPriceSuggestionsQuery extends MerchantQuery {
+
+	use ReportQueryTrait;
+
+	/**
+	 * MerchantPriceSuggestionsQuery constructor.
+	 *
+	 * @param array $args Query arguments.
+	 */
+	public function __construct( array $args ) {
+		parent::__construct( 'PriceInsightsProductView' );
+
+		$this->set_initial_columns();
+		$this->handle_query_args( $args );
+	}
+
+	/**
+	 * Filter the query by a list of product IDs.
+	 *
+	 * @param array $ids List of product IDs to filter by.
+	 *
+	 * @return $this
+	 */
+	public function filter( array $ids ): QueryInterface {
+		if ( ! empty( $ids ) ) {
+			$this->where( 'product_view.id', $ids, 'IN' );
+		}
+		return $this;
+	}
+
+	/**
+	 * Set the initial columns for this query.
+	 */
+	protected function set_initial_columns() {
+		$this->columns(
+			[
+				'id'                                    => 'product_view.id',
```

Comment:
Let's add offer_id here too
```suggestion
				'id'                                    => 'product_view.id',
				'offer_id'                              => 'product_view.offer_id',
```